packages/prerelease-registry/functions/gitHubFetch.ts

```diff
@@ -0,0 +1,23 @@
+export const generateGitHubFetch = ({
```

Comment:
Isn't it it a bit weird to have this file below the functions directory?
Intuitively I would then assume that `https://prerelease-registry.developers.workers.dev/gitHubFetch` was a valid route...

Reply:
Yeah, I understand. The same 'problem' exists in other projects today.

Generally, a purely static Pages project looks like this:

```txt
|-- src/
    |-- index.ts
    |-- ... other front-end code
|-- tsconfig.json
|-- ... other config related to the front-end code
```

We then added a `functions` directory, and generally, you'd want to put a different `tsconfig.json` in there:

```txt
|-- functions/
    |-- tsconfig.json
    |-- hello.ts
    |-- ... other API routes
|-- src/
    |-- index.ts
    |-- ... other front-end code
|-- tsconfig.json
|-- ... other config related to the front-end code
```

This means you generally can't 'escape' the `functions` directory. So I personally usually do something like this, which at least hints that these files don't export routes:

```txt
|-- functions/
    |-- utils/
        |-- fetcher.ts
    |-- tsconfig.json
    |-- hello.ts
    |-- ... other API routes
```

For this particular project, since it has no static front-end stuff, we can go up and out of `functions`. We could have a sibling `utils` or similar which we can put the shared non-route logic in. Or we could stick with something like what I've described above. Dealer's choice.

cc. @geelen (in case you haven't seen this 'problem' before). Making the `functions` directory configurable would also help, since we could set it to `/functions/routes` or similar:

```txt
|-- functions/
    |-- routes/
        |-- hello.ts
        |-- ... other API routes
    |-- utils/
        |-- fetcher.ts
    |-- tsconfig.json
```

Reply:
Thought about it a bit, and since we control this package entirely (including manually calling `wrangler pages functions build`), I think we can go straight to the `./functions/routes` option, which is pretty clean, imo.